common_markdown_files/DEVELOPMENT.md.erb

```diff
@@ -0,0 +1,246 @@
+or `byebug` for debugging---you can create a `Gemfile-custom` file containing those
+gem declarations. The `Gemfile` evaluates that file if it exists, and it is git-ignored.
+
+# Running the build
+
+The [Travis CI build](https://travis-ci.org/rspec/<%= project_name %>)
+runs many verification steps to prevent regressions and
+ensure high-quality code. To run the Travis build locally, run:
+
+```
+$ script/run_build
+```
+
+# What to Expect
+
+To ensure high, uniform code quality, all code changes (including
+changes from the maintainers!) are subject to a pull request code
+review. We'll often ask for clarification or suggest alternate ways
+to do things. Our code reviews are intended to be a two-way
+conversation.
+
+While every user-facing change needs a changelog entry, don't worry
+about adding it yourself. Since the changelog changes to frequently,
+it tends to be the source of merge conflicts when contributors include
+edits in their PRs, so we prefer to add changelog entries ourselves
+after merging your PR.
+
+# Adding Docs
+
+RSpec uses [YARD](http://yardoc.org/) for its API documentation. To
+ensure the docs render well, we recommend running a YARD server and
+viewing your edits in a browser.
+
+To run a YARD server:
+
+```
+$ bundle exec yard server --reload
+
+# or, if you installed your bundle with `--standalone --binstubs`:
+
+$ bin/yard server --reload
+```
+
+Then navigate to `localhost:8808` to view the rendered docs.
+
+# The CI build, in detail
+
+As mentioned above, RSpec runs many verification steps as part of its CI build.
+Let's break this down into the individual steps.
+
+## Specs
+
+RSpec dogfoods itself. It's primary defense against regressions is its spec suite. Run with:
+
+```
+$ bundle exec rspec
+
+# or, if you installed your bundle with `--standalone --binstubs`:
+
+$ bin/rspec
+```
+
+The spec suite performs a couple extra checks that are worth noting:
+
+* *That all the code is warning-free.* Any individual example that produces output
+  to `stderr` will fail. We also have a spec that loads all the `lib` and `spec`
+  files in a newly spawned process to detect load-time warnings and fail if there
+  are any. RSpec must be warning-free so that users who enable Ruby warnings will
+  not get warnings from our code.
+* *That only a minimal set of stdlibs are loaded.* Since Ruby makes loaded libraries
+  available for use in any context, we want to minimize how many bits of the standard
+  library we load and use. Otherwise, RSpec's use of part of the standard library could
+  mask a problem where a gem author forgets to load a part of the standard library they
+  rely on. The spec suite contains a spec that defines a whitelist of allowed loaded
+  stdlibs.
+
+In addition, we use [SimpleCov](https://github.com/colszowka/simplecov)
+to measure and enforce test coverage. If the coverage falls below a
+project-specific threshold, the build will fail.
+
+## Cukes
```

Comment:
Need to describe what is appropriate for cukes. Maybe:

* Cukes are our documentation. Only user facing examples that help demonstrate usage. Anything else should be in specs.
* Duplication between YARD and cukes is ok.